osquery/events/linux/auditdnetlink.cpp

```diff
@@ -857,32 +877,67 @@ bool AuditdNetlinkParser::ParseAuditReply(
 
   // Parse the record header
   event_record.type = reply.type;
-  boost::string_ref message_view(reply.message,
-                                 static_cast<unsigned int>(reply.len));
+  std::string_view message_view(reply.message,
+                                static_cast<std::size_t>(reply.len));
+
+  /* NOTE: The preamble is "audit(%llu.%03lu:%u): ",
```

Comment:
While I accept that this is a more careful parsing of the Audit message header, and the added comments are a big improvement ... is the use of some hardcoded constant offsets, and taking length measurements of the preamble, really the best way to parse it? What about a regex approach?